lib/modules/platform/codecommit/index.ts

```diff
@@ -61,52 +56,42 @@ export async function initPlatform({
   password,
   token: awsToken,
 }: PlatformParams): Promise<PlatformResult> {
-  let accessKeyId = username;
-  let secretAccessKey = password;
+  const accessKeyId = username;
+  const secretAccessKey = password;
   let region: string | undefined;
 
-  if (!accessKeyId) {
-    accessKeyId = process.env.AWS_ACCESS_KEY_ID;
+  if (accessKeyId) {
+    process.env.AWS_ACCESS_KEY_ID = accessKeyId;
   }
-  if (!secretAccessKey) {
-    secretAccessKey = process.env.AWS_SECRET_ACCESS_KEY;
+  if (secretAccessKey) {
+    process.env.AWS_SECRET_ACCESS_KEY = secretAccessKey;
+  }
+  if (awsToken) {
+    process.env.AWS_SESSION_TOKEN = awsToken;
   }
 
   if (endpoint) {
     const regionReg = regEx(/.*codecommit\.(?<region>.+)\.amazonaws\.com/);
     const codeCommitMatch = regionReg.exec(endpoint);
     region = codeCommitMatch?.groups?.region;
-    if (!region) {
+    if (region) {
+      process.env.AWS_REGION = region;
+    } else {
       logger.warn("Can't parse region, make sure your endpoint is correct");
     }
-  } else {
-    region = process.env.AWS_REGION;
-  }
-
-  if (!accessKeyId || !secretAccessKey || !region) {
-    throw new Error(
-      'Init: You must configure a AWS user(accessKeyId), password(secretAccessKey) and endpoint/AWS_REGION'
-    );
   }
 
-  config.region = region;
-  const credentials: Credentials = {
-    accessKeyId,
-    secretAccessKey,
-    sessionToken: awsToken ?? process.env.AWS_SESSION_TOKEN,
-  };
-  config.credentials = credentials;
-
   // If any of the below fails, it will throw an exception stopping the program.
-  client.buildCodeCommitClient(region, credentials);
-  // To check if we have permission to codecommit
+  client.buildCodeCommitClient();
+  // To check if we have permission to codecommit, throws exception if failed.
   await client.listRepositories();
 
-  initIamClient(region, credentials);
-  config.userArn = await getUserArn();
-
   const platformConfig: PlatformResult = {
-    endpoint: endpoint ?? `https://git-codecommit.${region}.amazonaws.com/`,
+    endpoint:
+      endpoint ??
+      `https://git-codecommit.${
+        process.env.AWS_REGION ?? 'us-east-1'
+      }.amazonaws.com/`,
```

Comment:
`endpoint` isn't mandatory?

Reply:
I added it because u asked for it in one of the old PRs. Im not sure why we return it or if it is used by the abstract code

Reply:
I thought `endpoint` was a mandatory config option for Renovate here, so this would not be executed